src/script_lua.c

```diff
@@ -819,8 +819,15 @@ static robj **luaArgsToRedisArgv(lua_State *lua, int *argc, int *argv_len) {
             /* We can't use lua_tolstring() for number -> string conversion
              * since Lua uses a format specifier that loses precision. */
             lua_Number num = lua_tonumber(lua,j+1);
-            obj_len = fpconv_dtoa((double)num, dbuf);
-            dbuf[obj_len] = '\0';
+            /* Integer printing function is much faster, check if we can safely use it.
+             * This also avoids converting 100000000 to 1e9. */
```

Comment:
>This also avoids converting 100000000 to 1e9

but does this really solves the problem you're after?
if the parsing code can't handle scientific notation, it'll still be unable to handle some numbers returned by fpconv_dtoa.

Reply:
ok, now i get it.
```suggestion
             * Since lua_Number is not explicitly an integer or a double, we need to make
             *  an effort to convert it to an integer when that's possible, since it could be used
             * in a context that doesn't support scientific notation (e.g. 1e9 instead of 100000000). */
```